File: scheduler/cassandra-persistence/src/main/java/org/conductoross/conductor/scheduler/cassandra/config/CassandraSchedulerConfiguration.java
package org.conductoross.conductor.scheduler.cassandra.config;

import org.conductoross.conductor.scheduler.cassandra.dao.CassandraSchedulerArchivalDAO;
import org.conductoross.conductor.scheduler.cassandra.dao.CassandraSchedulerDAO;
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.netflix.conductor.cassandra.config.CassandraProperties;

import com.datastax.driver.core.Session;
import com.fasterxml.jackson.databind.ObjectMapper;
import io.orkes.conductor.dao.archive.SchedulerArchivalDAO;
import io.orkes.conductor.dao.scheduler.SchedulerDAO;

@Configuration(proxyBeanMethods = false)
@ConditionalOnProperty(name = "conductor.db.type", havingValue = "cassandra")
public class CassandraSchedulerConfiguration {

    @Bean
    @ConditionalOnProperty(
            name = "conductor.scheduler.enabled",
            havingValue = "true",
            matchIfMissing = false)
    public SchedulerDAO cassandraSchedulerDAO(
            Session session, ObjectMapper objectMapper, CassandraProperties properties) {
        return new CassandraSchedulerDAO(session, objectMapper, properties);
    }

    @Bean
    @ConditionalOnProperty(
            name = "conductor.scheduler.enabled",
            havingValue = "true",
            matchIfMissing = false)
    public SchedulerArchivalDAO cassandraSchedulerArchivalDAO(
            Session session, ObjectMapper objectMapper, CassandraProperties properties) {
        return new CassandraSchedulerArchivalDAO(session, objectMapper, properties);
    }
}

File: scheduler/cassandra-persistence/src/main/typescript/org/conductoross/conductor/scheduler/cassandra/config/CassandraSchedulerConfiguration.ts
import { CassandraSchedulerArchivalDAO } from '../dao/CassandraSchedulerArchivalDAO';
import { CassandraSchedulerDAO } from '../dao/CassandraSchedulerDAO';
import { CassandraProperties } from '../../../../../../../../../../mock';
import { Client } from 'cassandra-driver';
import { SchedulerArchivalDAO } from '../../../../../../../../../../core/src/main/typescript/io/orkes/conductor/dao/archive/SchedulerArchivalDAO';
import { SchedulerDAO } from '../../../../../../../../../../core/src/main/typescript/io/orkes/conductor/dao/scheduler/SchedulerDAO';

export class CassandraSchedulerConfiguration {
    public cassandraSchedulerDAO(
        session: Client,
        properties: CassandraProperties
    ): SchedulerDAO {
        return new CassandraSchedulerDAO(session, properties);
    }

    public cassandraSchedulerArchivalDAO(
        session: Client,
        properties: CassandraProperties
    ): SchedulerArchivalDAO {
        return new CassandraSchedulerArchivalDAO(session, properties);
    }
}
